Clockwork/Request/ShouldCollect.php

```diff
@@ -78,7 +78,7 @@ protected function passExcept(IncomingRequest $request)
 		if (! count($this->except)) return true;
 
 		foreach ($this->except as $pattern) {
-			if (preg_match('#' . str_replace('#', '\#', $pattern) . '#', $request->uri)) return false;
+			if (preg_match('#' . preg_quote($pattern) . '#', $request->uri)) return false;
```

Comment:
Pass delimiter: `preg_quote($pattern, '#')`, though in this case # will automatically be quoted anyway.

Reply:
In my case I want to quote a question symbol: '?'

Actual use case: I have 'api' config option:
`'api' => 'index.php?a=Clockwork&request='`

Then in source code I that api url automatically ignored - this is cool, but does not work!
https://github.com/itsgoingd/clockwork/blob/543de6b1c17021183be5fd2dd6c7ffa153d9ab2d/Clockwork/Support/Vanilla/Clockwork.php#L242
```
// don't collect data for Clockwork requests
$this->clockwork->shouldCollect()->except(rtrim($this->config['api'], '/'));
```

Why it does not work?
Because `preg_match('#index.php?a=Clockwork&request=#', $request->uri)` returns 0 even when `$request->uri === '/index.php?a=Clockwork&request=12345'`.

So we need to escape **?** symbol (and many others special regex symbols) in the first argument of `preg_match`

Reply:
https://www.php.net/manual/en/function.preg-match.php

According to the doc, the second parameter should have the pattern delimiter, which is '#' as `preg_match('#XXX#')` starts and ends with #. Yes, in this case it can be left out, but I usually consider missing delimiter being a bug when I quickly browse through the code.